cdap-common/src/main/java/io/cdap/cdap/common/http/AuthenticationChannelHandler.java

```diff
@@ -86,27 +91,21 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
             throw e;
           }
         }
-      } else if (enforceAuthenticatedRequests) {
-        throw new IllegalArgumentException(String.format("Missing Authorization header for request from IP %s",
-                                                         ctx.channel().remoteAddress().toString()));
       }
-
-      SecurityRequestContext.setUserId(currentUserId);
+      LOG.trace("Got user ID '{}' user IP '{}' from IP '{}' and authorization header length '{}'",
+                userID == null ? "NULL" : userID,
```

Comment:
I think you don't need the `: ?`. Having a `null` value would resulted in a `null` string in the log text.

Reply:
Fixed